lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -341,8 +341,14 @@ def access_token_url
 
       def setup_access_token
         request = 'grant_type=client_credentials'
-        response = parse(ssl_post(access_token_url, request, access_token_header))
-        response['access_token']
+        begin
+          raw_response = ssl_post(access_token_url, request, access_token_header)
+        rescue ResponseError => e
+          raise OAuthResponseError.new(e)
+        else
+          response = parse(raw_response)
+          response['access_token']
```

Comment:
Should we raise the `OAuthResponseError` if `response['access_token']` is not present?

Reply:
Yeah that's a good idea, will change it.